### ISSUE TITLE
Storybook sidebar navigation: aria-expanded should be on button, not parent

### ISSUE THREAD

**Issue by micahgodbolt** [storybookjs/storybook#18346]
aria-expanded is meant to be used with button, or some other role. It is currently be added to the parent div with no role making the attribute ineffective.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#associated_roles

Moving the aria-expanded to the button that does the expanding will provide better accessibility to those using screen readers.

![image](https://user-images.githubusercontent.com/1434956/170762084-ec6c776a-a4c5-4807-8aea-3147aca4f2ef.png)

**Comment by micahgodbolt** [comment]
Looks like this is usually a button in the case of stories

![image](https://user-images.githubusercontent.com/1434956/170763061-b86a9ce9-06a1-4f54-a6c8-90447048cd9d.png)

**Comment by micahgodbolt** [comment]
Looks like the problem is right here

https://github.com/storybookjs/storybook/blob/f74a04a411c438880b2f08e4e7a5e8cc8b2359f9/lib/ui/src/components/sidebar/Tree.tsx#L155

This should be moved down to the CollapseButton

**Comment by Daraphista** [comment]
Hi, I can help out. Could you assign this issue to me?

**Comment by ithiel883** [comment]
can i be assigned this issue?

**Comment by unknownbear** [comment]
hahah im not competent but for what its w0rthl haha, it might help me personally

**Comment by shilman** [comment]
Gadzooks!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.5.10-alpha.0 containing PR #18354 that references this issue. Upgrade today to the `@prerelease` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

**Comment by shilman** [comment]
Son of a gun!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.5.10 containing PR #18354 that references this issue. Upgrade today to the `@latest` NPM tag to try it out!

```
npx sb upgrade
```
